src/main/java/se/liu/ida/hefquin/engine/wrappers/graphqlwrapper/impl/SPARQL2GraphQLTranslatorImpl.java

```diff
@@ -0,0 +1,454 @@
+        
+        int entrypointCounter = 0;
+        final Set<String> objectTypeNames = endpoint.getGraphQLObjectTypes();
+        for(final String objectTypeName : objectTypeNames){
+            // Get the full list entrypoint
+            final GraphQLEntrypoint e = endpoint.getEntrypoint(objectTypeName,GraphQLEntrypointType.FULL);
+            final StringBuilder currentPath = new StringBuilder("ep_full")
+            .append(entrypointCounter).append(":").append(e.getFieldName()).append("/");
+            fieldPaths.add(new StringBuilder(currentPath).append("id_").append(objectTypeName).append(":id").toString());
+
+            final Set<String> allObjectURI = URI2GraphQLHelper.getPropertyURIs(objectTypeName,GraphQLFieldType.OBJECT,config, endpoint);
+            final Set<String> allScalarURI = URI2GraphQLHelper.getPropertyURIs(objectTypeName,GraphQLFieldType.SCALAR,config, endpoint);
+
+            // Add all fields that nests another object
+            for(final String uri : allObjectURI){
+                addEmptyObjectField(fieldPaths, subgraphPatterns, 
+                    connectors, config, endpoint, currentPath.toString(), objectTypeName, uri);
+            }
+            // Add all fields that represent scalar values
+            for(final String uri : allScalarURI){
+                addScalarField(fieldPaths,config,currentPath.toString(),uri);
+            }
+
+            ++entrypointCounter;
+        }
+    }
+
+    /**
+     * Helper function which checks if any triple pattern predicate in @param sgp is a variable or blank node, returns true if so
+     */
+    protected boolean hasVariablePredicate(final Set<TP> sgp){
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            if(predicate.isVariable()){
+                return true;
+            }    
+        }
+        return false;
+    }
+
+    /**
+     * Returns the GraphQL object type @param sgp correponds to. 
+     * If the type is undeterminable @return null
+     */
+    protected String determineSgpType(final Set<TP> sgp, final GraphQL2RDFConfiguration config){
+        for(final TP t : sgp){
+            final Node predicate = t.getTriplePattern().asJenaTriple().getPredicate();
+            final Node object = t.getTriplePattern().asJenaTriple().getObject();
+
+            if(predicate.isURI() && predicate.getURI().startsWith(config.getPropertyPrefix())){
+                return config.getClassFromPropertyURI(predicate.getURI());
+            }
+            else if(predicate.isURI() && predicate.getURI().equals(config.getClassMembershipURI())){
+                if(object.isURI() && object.getURI().startsWith(config.getClassPrefix())){
+                    final int splitIndex = config.getClassPrefix().length();
+                    final String sgpType = object.getURI().substring(splitIndex);
+                    return sgpType;
```

Comment:
directly return the computed value, no need to store it in a variable before returning